Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm

```diff
@@ -66,15 +66,15 @@
 
 inline static bool shouldBlockTrackersForThirdPartyCloaking(NSURLRequest *request)
 {
-    auto requestURL = request.URL;
-    auto mainDocumentURL = request.mainDocumentURL;
+    RetainPtr<NSURL> requestURL = request.URL;
+    RetainPtr<NSURL> mainDocumentURL = request.mainDocumentURL;
```

Comment:
Can this be just `RetainPtr`?

Reply:
Everywhere I added the explicit template parameter is because the compiler required me to.

Reply:
I wonder why the template deduction guide doesn't help here:

`template<typename T> RetainPtr(T) -> RetainPtr<RetainPtrType<T>>;`

`request.URL` is NSURL*.

Reply:
I don't know. It does work *sometimes*. But in quite a few places, it forced me to specify the type and I am not sure why.